### DISCUSSION TITLE
Where is generated C code?

### DISCUSSION THREAD

**Discussion by ghost** [edubart/nelua-lang#177] [Q&A] [answered]
I'm thinking about making a game engine and using nelua as the scripting language, where is the C code generated? when running `nelua main.nelua`, no files are generated in the cwd

**Comment by Andre-LA** [comment, marked as answer]
The C code is generated at a `cache` directory. In the case of Linux, it's at `~/.cache/nelua/myprogram.c`.

**Comment by IcedQuinn** [comment]
Might be worth having a flag to leave the .c file in the same directory as the .nelua. Tup (a build system) used to complain about cache directories (though I think they fixed it so writing to dot folders doesn't count now.)

Although it looks like `nelua --print-code something.nelua > something.c` may do that.

**Reply by Its-Kenta** [reply]
Correct me if I'm wrong, but wasn't there a way to change that using .neluacfg?

**Reply by Andre-LA** [reply]
You can with the `--output` (or just `-o`) flag, like `nelua mycode.nelua -o result.c`

**Comment by stefanos82** [comment]
> I'm thinking about making a game engine and using nelua as the scripting language, where is the C code generated? when running `nelua main.nelua`, no files are generated in the cwd

The best approach in your case would be to use `nelua --cache-dir . main.nelua`; it literally means `use this current working directory as my cache directory`.
